cmd/thanos/downsample.go

```diff
@@ -158,15 +159,18 @@ func downsampleBucket(
 	bkt objstore.Bucket,
 	metas map[ulid.ULID]*metadata.Meta,
 	dir string,
-) error {
-	if err := os.RemoveAll(dir); err != nil {
-		return errors.Wrap(err, "clean working directory")
-	}
+	hashFunc metadata.HashFunc,
+) (rerr error) {
 	if err := os.MkdirAll(dir, 0777); err != nil {
 		return errors.Wrap(err, "create dir")
 	}
 
 	defer func() {
+		// Leave the downsample directory for inspection if it is a halt error
+		// or if it is not then so that possibly we would not have to download everything again.
+		if rerr != nil {
```

Comment:
What if it's disk full issue?

Reply:
Then it's not a problem because we have a consistent order of work with https://github.com/thanos-io/thanos/pull/3843 in place + https://github.com/thanos-io/thanos/blob/6877bd17aa3ebee44304aee4b2c6582be6f860cc/pkg/compact/compact.go#L312-L314 this exists already. This means that we are sure that after a retry, we are going to restart the work on the same blocks. If the blocks would disappear then accordingly we would get rid of them and then Thanos Compactor can continue its work. Does it make sense, @bwplotka ?

Reply:
Makes sense 👍🏽